pandas/core/generic.py

```diff
@@ -5907,6 +5908,137 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _as_nullable_type(self: ABCSeries) -> ABCSeries:
+        """
+        Handle one Series
+
+        Rules:
+        If an object, see if we can infer string, boolean or integer, otherwise leave
+        alone
+        If an integer and not an extension type, convert to the Int64/Int32 type
+        (platform dependent)
+        If numeric, see if we can infer integer, otherwise try to use astype() to make
+        it integer.
+
+        """
+        dtype = self.dtype
+        new_dtype = dtype
+        changeit = False
+        result = self
+        target_int_dtype = "Int64"
+        if is_platform_32bit():
+            target_int_dtype = "Int32"
```

Comment:
Is this platform distinction needed? 
I thought that in our constructors like `Series(..)` we always inferred int64 (independent from the platform, in contrast to eg numpy's `np.array([..])`). If that is the case, I think we can do the same here.

Reply:
I was wondering about that.  I'm going to remove that, as I've no way to easily test it!